### PR TITLE
Fix: stake limit resume handling

### PR DESCRIPTION
**Pull request by TheDZhon**
image

---

Test cases are in [`test/0.4.24/staking-limit.test.js`](https://github.com/lidofinance/lido-dao/blob/59b0476fef2c89a09e3b7cca264923eb6f9635b9/test/0.4.24/staking-limit.test.js#L244-L286)

### CODE REVIEW

File: contracts/0.4.24/lib/StakeLimitUtils.sol [hunk 1]
```diff
@@ -145,13 +145,19 @@ library StakeLimitUtils {
...
-        // or new limit is lower than previous, then
-        // reset prev stake limit to the new max stake limit
-        if ((_data.maxStakeLimit == 0) || (_maxStakeLimit < _data.prevStakeLimit)) {
+        // reset prev stake limit to the new max stake limit if
+        if (
+            // staking was paused or
+            _data.prevStakeBlockNumber == 0 ||
+            // staking was unlimited or
+            _data.maxStakeLimit == 0 ||
+            // new limit is lower than the previous
```
Comment (folkyatina) [inline contracts/0.4.24/lib/StakeLimitUtils.sol:154]: Comment is misleading here. It make you think that prevStakeLimit is prevMaxStakeLimit, but it's a current level in the Diablo mana sphere

**Review by folkyatina — APPROVED** [review]
Fine, but can optionally use some  work on naming and comments. 
As for me, `prevStakeLimit` is really misleading name here and can use some rebranding.

**Review by loga4 — APPROVED** [review]
yep, all is good

File: contracts/0.4.24/lib/StakeLimitUtils.sol (hunk 1 above)
Comment (TheDZhon) [reply contracts/0.4.24/lib/StakeLimitUtils.sol]: yeah, good catch!
ty, fixed :+1:

**Comment by TheDZhon** [comment]
> Fine, but can optionally use some work on naming and comments. As for me, `prevStakeLimit` is really misleading name here and can use some rebranding.

Agree, added a few comments, but decided to not proceed with a large-scale refactoring

**Pull request merged**